src/Components/BehaviorsModal/Internal/Widgets/ValueWidget/ValueWidget.tsx

```diff
@@ -34,26 +37,35 @@ export const ValueWidget: React.FC<IProp> = ({ widget, placeholderValues }) => {
     const theme = useTheme();
     const { t, i18n } = useTranslation();
 
-    const { twins } = useContext(BehaviorsModalContext);
+    const { twins, mode } = useContext(BehaviorsModalContext);
     const { displayName, valueExpression, type } = widget.widgetConfiguration;
 
-    const parsedValue = useMemo(
-        () => parseLinkedTwinExpression(valueExpression, twins),
-
-        [valueExpression, twins]
-    );
+    const parseResult: { value: any; isValid: boolean } = useMemo(() => {
+        const parsedValue = parseLinkedTwinExpression(valueExpression, twins);
+        if (parsedValue === '') {
+            if (mode === BehaviorModalMode.viewer) {
+                return { value: null, isValid: false };
+            } else {
+                return {
+                    value:
+                        placeholderValues?.[type] ??
+                        getValuePlaceholders(t)[type],
+                    isValid: true
+                };
+            }
+        } else {
+            return { value: parsedValue, isValid: true };
+        }
+    }, [valueExpression, twins, mode, placeholderValues, type, t]);
 
     const styles = getStyles(theme);
     return (
         <div className={styles.container}>
             <div className={styles.expressionValueContainer}>
                 <FormattedValue
                     locale={i18n.language as Locale}
-                    value={
-                        parsedValue ||
-                        (placeholderValues?.[type] ??
```

Comment:
this seems like now we won't pass the placeholder value ever?

Reply:
Moved this logic into the useMemo